addons/im_livechat/static/src/core/common/thread_model_patch.js

```diff
@@ -77,4 +78,56 @@ patch(Thread.prototype, {
         }
         return super.getPersonaName(persona);
     },
+    get displayName() {
+        if (this.channel?.channel_type !== "livechat" || this.self_member_id?.custom_channel_name) {
+            return super.displayName;
+        }
+        const memberType = this.self_member_id?.livechat_member_type;
+        if (memberType === "visitor") {
+            const agents = this.correspondents.filter((c) => c.livechat_member_type === "agent");
+            if (agents.length) {
+                return formatList(agents.map((a) => this.getPersonaName(a.partner_id)));
+            }
+            if (this.channel?.livechat_agent_history_ids.length) {
+                return formatList(
+                    this.channel.livechat_agent_history_ids.map((h) =>
+                        this.getPersonaName(h.partner_id)
+                    )
+                );
+            }
+        }
+        if (memberType === "agent") {
+            const visitors = this.correspondents.filter(
+                (c) => c.livechat_member_type === "visitor"
+            );
+            if (visitors.length) {
+                return formatList(
+                    visitors.map((v) => (v.partner_id ? v.partner_id.name : v.guest_id.name))
+                );
+            }
+            if (this.channel?.livechat_customer_history_ids.length) {
+                return formatList(
+                    this.channel?.livechat_customer_history_ids.map((h) =>
+                        h.partner_id ? h.partner_id.name : h.guest_id.name
+                    )
+                );
```

Comment:
Should use `getPersonaName` whenever possible.

`partner_id.name` should never be used explicitly. `partner.displayName` is always better and even better is message.authorName, member.name or getPersonaName